auditlog/history.py

```diff
@@ -0,0 +1,131 @@
+import datetime
+from dataclasses import dataclass
+from typing import Any, Optional
+
+
+@dataclass
+class HistoricalObjectState:
+    log_found: bool
+    """Boolean representing the presence of a log for the date given"""
+    serialized_fields: Optional[dict] = None
+    """Serialized fields from the data on the log entry or none if log not found"""
+    timestamp: Optional[datetime.datetime] = None
+    """Timestamp of the log for this state or none if log not found"""
+    log_entry_id: Optional[int] = None
+    """ID of the applicable LogEntry record"""
+
+    @property
+    def log_entry(self):
+        from .models import LogEntry
+
+        return LogEntry.objects.get(id=self.log_entry_id)
+
+
+@dataclass
+class HistoricalFieldState:
+    log_found: bool
+    """Boolean representing the presence of a log for the date given"""
+    field_found: bool
+    """Boolean representing the presence of a field in the serialized data."""
+    field_name: str
+    """Name of the field being referenced"""
+    value: Optional[Any] = None
+    """Value of the found field or none if not found"""
+    timestamp: Optional[datetime.datetime] = None
+    """Timestamp of the log for this state of the field value or none if not found"""
+    log_entry_id: Optional[int] = None
+    """ID of the applicable LogEntry record"""
+
+    @property
+    def log_entry(self):
+        from .models import LogEntry
+
+        return LogEntry.objects.get(id=self.log_entry_id)
+
+
+class HistoricalStateLookupManagerMixin:
+    def get_for_object(self, instance):
+        raise NotImplementedError(f"{self.__class__} requires a get_for_object method.")
+
+    def get_object_state_at_timestamp(self, instance, timestamp):
```

Comment:
Why we don't return the found log entry in this function instead of `HistoricalObjectState`?

What would be the benefit of `HistoricalObjectState` and `HistoricalObjectState`?

Reply:
removed.